From 2e0fd82993a77f6d46b6522aea5c76c2abc710fb Mon Sep 17 00:00:00 2001
From: PyScotch <pyscotch@noreply.invalid>
Date: Tue, 13 Oct 2026 16:06:22 +0000
Subject: [PATCH 209/240] Remove unused `common_error.c` from libScotch [report
 PyScotch]

"common_error.c" is a leftover of the time when the error handling
routines were part of the library itself. The Makefile of libScotch
has never compiled it: the error routines are provided by the
libScotchErr and libScotchErrExit libraries, and libScotch only refers
to `SCOTCH_errorPrint()` and `SCOTCH_errorPrintW()`.

However, "src/libscotch/CMakeLists.txt" still listed it, so that a
CMake-built libScotch exported the un-prefixed global symbols
`errorPrint()`, `errorPrintW()` and `errorProg()`, which are never
called by the library and may collide with routines of user programs.

Remove the file and its CMake entry.

Refs: BLD-01
---
 src/libscotch/CMakeLists.txt |   1 -
 src/libscotch/common_error.c | 123 -----------------------------------
 2 files changed, 124 deletions(-)
 delete mode 100644 src/libscotch/common_error.c

diff --git a/src/libscotch/CMakeLists.txt b/src/libscotch/CMakeLists.txt
index c50981b..4c6ffdc 100644
--- a/src/libscotch/CMakeLists.txt
+++ b/src/libscotch/CMakeLists.txt
@@ -207,7 +207,6 @@ set(SCOTCH_C_SOURCES
   common.c
   common.h
   common_context.c
-  common_error.c
   common_file.c
   common_file.h
   common_file_compress.c
diff --git a/src/libscotch/common_error.c b/src/libscotch/common_error.c
deleted file mode 100644
index 76b2dc4..0000000
--- a/src/libscotch/common_error.c
+++ /dev/null
@@ -1,123 +0,0 @@
-/* Copyright 2004,2007,2023 IPB, Universite de Bordeaux, INRIA & CNRS
-**
-** This file is part of the Scotch software package for static mapping,
-** graph partitioning and sparse matrix ordering.
-**
-** This software is governed by the CeCILL-C license under French law
-** and abiding by the rules of distribution of free software. You can
-** use, modify and/or redistribute the software under the terms of the
-** CeCILL-C license as circulated by CEA, CNRS and INRIA at the following
-** URL: "http://www.cecill.info".
-**
-** As a counterpart to the access to the source code and rights to copy,
-** modify and redistribute granted by the license, users are provided
-** only with a limited warranty and the software's author, the holder of
-** the economic rights, and the successive licensors have only limited
-** liability.
-**
-** In this respect, the user's attention is drawn to the risks associated
-** with loading, using, modifying and/or developing or reproducing the
-** software by the user in light of its specific status of free software,
-** that may mean that it is complicated to manipulate, and that also
-** therefore means that it is reserved for developers and experienced
-** professionals having in-depth computer knowledge. Users are therefore
-** encouraged to load and test the software's suitability as regards
-** their requirements in conditions enabling the security of their
-** systems and/or data to be ensured and, more generally, to use and
-** operate it in the same conditions as regards security.
-**
-** The fact that you are presently reading this means that you have had
-** knowledge of the CeCILL-C license and that you accept its terms.
-*/
-/************************************************************/
-/**                                                        **/
-/**   NAME       : common_error.c                          **/
-/**                                                        **/
-/**   AUTHORS    : David GOUDIN                            **/
-/**                Pascal HENON                            **/
-/**                Francois PELLEGRINI                     **/
-/**                Pierre RAMET                            **/
-/**                                                        **/
-/**   FUNCTION   : Part of a parallel direct block solver. **/
-/**                This module handles errors.             **/
-/**                                                        **/
-/**   DATES      : # Version 0.0  : from : 08 may 1998     **/
-/**                                 to   : 02 oct 1998     **/
-/**                # Version 7.0  : from : 19 jan 2023     **/
-/**                                 to   : 19 jan 2023     **/
-/**                                                        **/
-/************************************************************/
-
-/*
-**  The defines and includes.
-*/
-
-#include "common.h"
-
-/********************************/
-/*                              */
-/* The error handling routines. */
-/*                              */
-/********************************/
-
-static char                 errorProgName[32] = "";
-
-/* This routine sets the program name for
-** error reporting.
-** It returns:
-** - void  : in all cases.
-*/
-
-void
-errorProg (
-const char * const          progstr)              /*+ Program name +*/
-{
-  strncpy (errorProgName, progstr, 29);
-  errorProgName[29] = '\0';
-  strcat  (errorProgName, ": ");
-
-}
-
-/* This routine prints an error message with
-** a variable number of arguments, as printf ()
-** does, and exits.
-** It returns:
-** - EXIT  : in all cases.
-*/
-
-void
-errorPrint (
-const char * const          errstr,               /*+ printf-like variable argument list */
-...)
-{
-  va_list             errlist;                    /* Argument list of the call */
-
-  fprintf  (stderr, "%sERROR: ", errorProgName);
-  va_start (errlist, errstr);
-  vfprintf (stderr, errstr, errlist);             /* Print arguments */
-  va_end   (errlist);
-  fprintf  (stderr, "\n");
-  fflush   (stderr);                              /* In case it has been set to buffered mode */
-}
-
-/* This routine prints a warning message with
-** a variable number of arguments, as printf ()
-** does.
-** It returns:
-** - void  : in all cases.
-*/
-
-void
-errorPrintW (
-const char * const          errstr,               /*+ printf-like variable argument list */
-...)
-{
-  va_list             errlist;                    /* Argument list of the call */
-
-  fprintf  (stderr, "%sWARNING: ", errorProgName);
-  va_start (errlist, errstr);
-  vfprintf (stderr, errstr, errlist);             /* Print arguments */
-  va_end   (errlist);
-  fprintf  (stderr, "\n");
-  fflush   (stderr);                              /* In case it has been set to buffered mode */
-}
-- 
2.47.3

